hibernate-core/src/main/java/org/hibernate/mapping/SimpleValue.java

```diff
@@ -434,7 +437,27 @@ public RootClass getRootClass() {
 
 		// default initial value and allocation size per-JPA defaults
 		params.setProperty( OptimizableGenerator.INITIAL_PARAM, String.valueOf( OptimizableGenerator.DEFAULT_INITIAL_VALUE ) );
-		params.setProperty( OptimizableGenerator.INCREMENT_PARAM, String.valueOf( OptimizableGenerator.DEFAULT_INCREMENT_SIZE ) );
+		final ConfigurationService cs = metadata.getMetadataBuildingOptions().getServiceRegistry()
+				.getService( ConfigurationService.class );
+
+		final String idNamingStrategy = cs.getSetting(
+				AvailableSettings.ID_DB_STRUCTURE_NAMING_STRATEGY,
+				StandardConverters.STRING,
+				null
+		);
+
+		if ( LegacyNamingStrategy.STRATEGY_NAME.equals( idNamingStrategy )
```

Comment:
To me, this just does not feel right.  Shouldn't this be based on (1) `hbm.xml` versus (2) annotations/`orm.xml`?

Reply:
I added tests,may be I'm missing something case, but when using `LegacyNamingStrategy.STRATEGY_NAME` I tried to obtain the same increment as we have in 5.X

Reply:
So do you have a test that shows any combination of settings that causes annotation / `orm.xml` mappings to use increment of 1 on 5.x?

Reply:
for a few cases https://github.com/hibernate/hibernate-orm/pull/4920/files#diff-3fc2829c3bdfbe2d7f4b9a2206d286f3fe586a145df31b773d444ce3e614f1ca

Reply:
not sure it covers enough cases

Reply:
Can you point me to the specific test that uses annotations (or `orm.xml`) and achieves increment = 1 (without actually configuring increment) on 5.x?

Reply:
https://github.com/hibernate/hibernate-orm/pull/4920/files#diff-3fc2829c3bdfbe2d7f4b9a2206d286f3fe586a145df31b773d444ce3e614f1caR63
and
https://github.com/hibernate/hibernate-orm/pull/4920/files#diff-3fc2829c3bdfbe2d7f4b9a2206d286f3fe586a145df31b773d444ce3e614f1caR83
and
https://github.com/hibernate/hibernate-orm/pull/4920/files#diff-3fc2829c3bdfbe2d7f4b9a2206d286f3fe586a145df31b773d444ce3e614f1caR143